docs/data-modeling/field-types.md

```diff
@@ -1,152 +1,189 @@
-- **Foreign Key**: The column in this table that (usually) refers to the entity key of another table in order to connect data from different tables that are related. For example, in a Products table, you might have a Customer ID field that points to a Customers table, where Customer ID is the primary key.
+## Data types
 
-## Common
+Data types are the underlying column types as defined in your database. Metabase reads the data types during the [database sync process](../databases/sync-scan.md). Because Metabase connects to many different databases, it uses its own type hierarchy under the hood, so that it could offer the same functionality for, for example, date fields in MongoDB or PostgreSQL
 
-Common field types are text/varchar types.
+The main data types in Metabase:
 
-- **Category**: If set to category, Metabase will know that it can use this field to group results when creating questions for automatic insights like [X-rays](../exploration-and-organization/x-rays.md).
-- **Comment**
-- **Description**
-- **Title**
+| Data Type  | Example database types                      |
+| ---------- | ------------------------------------------- |
+| Numeric    | `INTEGER`, `FLOAT`                          |
+| Temporal   | `DATE`, `TIMESTAMP`                         |
+| Text       | `VARCHAR`, `TEXT`                           |
+| Text-like  | MongoDB `BSONID`, Postgres `Enum`           |
+| Boolean    | Boolean                                     |
+| Collection | `JSON`, BigQuery `RECORD`, MongoDB `Object` |
 
-## Location
+Metabase currently does not support array types with any database. You'll only be able to use **Is empty** or **Is not empty** filters on columns containing arrays.
 
-Marking fields as locations tell Metabase that it can use the field to display data on a [map chart](../questions/visualizations/map.md).
+For some fields, you tell Metabase to [cast them to a different data type](#editing-data-and-semantic-types) (for example, text to date).
 
-- **City**
-- **Country**
-- **Latitude**: Tells Metabase that the data could be plotted on a [pin map](../questions/visualizations/map.md).
-- **Longitude**: Did you know that you can use the moons of Jupiter to calculate longitude?
-- **State**
-- **Zip Code**
+## Semantic types
 
-## Financial
+Semantic types are extra flavor that you can add to a field to communicate meaning and enable [additional functionality](#additional-functionality-for-data-and-semantic-types). Available semantic types depend on the underlying data types.
 
-Financial field types tell Metabase to treat the field's values as money. If you select any of the financial field types, Metabase will ask which currency you want to use. Metabase treats each of the following types as money, their differences are purely semantic.
+### Semantic types for any field
```

Comment:
This section could use some editing. E.g., line items don't need line breaks; category description is missing.